### PR TITLE
Conditions Reordering - sideeffects fix

### LINKED ISSUE #3322: [Conditionals] Description displays incorrect condition on condition re-order
## Overview
Descriptions are not following their conditions on re-order.

## Expected Behavior
Descriptions should match the condition being edited after being re-ordered.

## Steps to Reproduce
1. Create a condition set with two conditions
1. Swap the order of the conditions
1. Edit the second condition
1. Verify the description of the first condition changes on edit of second condition set

condition reorder

### PR DESCRIPTION
**Pull request by shefalijoshi**
Resolves #3322 
Ensures the right conditions are being updated and removed after reordering by using id rather than index to find conditions

### Author Checklist
1. Changes address original issue? Yes
2. Unit tests included and/or updated with changes? Yes
3. Command line build passes? Yes
4. Changes have been smoke-tested? Yes
5. Testing instructions included? Yes

### CODE REVIEW

File: src/plugins/condition/ConditionManager.js [hunk 1]
```diff
@@ -220,8 +232,8 @@ export default class ConditionManager extends EventEmitter {
         reorderPlan.forEach((reorderEvent) => {
             let item = oldConditions[reorderEvent.oldIndex];
             newCollection.push(item);
-            this.conditionSetDomainObject.configuration.conditionCollection = newCollection;
         });
+        this.conditionSetDomainObject.configuration.conditionCollection = newCollection;
```
Comment (davetsay) [inline src/plugins/condition/ConditionManager.js:236]: yeah, nice. this is much better.

File: src/plugins/condition/ConditionManager.js [hunk 2]
```diff
@@ -202,12 +208,18 @@ export default class ConditionManager extends EventEmitter {
...
-        this.conditionSetDomainObject.configuration.conditionCollection.splice(index, 1);
-        this.persistConditions();
+    removeCondition(id) {
+        let index = this.conditions.findIndex(item => item.id === id);
+        if (index > -1) {
+            this.conditions[index].destroy();
+            this.conditions.splice(index, 1);
+        }
+
+        index = this.conditionSetDomainObject.configuration.conditionCollection.findIndex(item => item.id === id);
```
Comment (davetsay) [inline src/plugins/condition/ConditionManager.js:218]: I'd use a separate variable definition for condition index and conditionCollection index.

Comment (shefalijoshi) [reply src/plugins/condition/ConditionManager.js]: Done.

**Comment by shefalijoshi** [comment]
> Looks good and I smoke tested. I have a couple comments in line.
> 
> Also, not your doing here, but I see in `condition.vue` we use e, ev, and event. Can we use one name?

Done.

**Pull request merged**